server/models/user.js

```diff
@@ -37,22 +37,37 @@ UserSchema.plugin(uniqueValidator, {
 UserSchema.pre('save', function(next) {
   const password = this.password;
   if (password) {
-    bcrypt.hash(password, 10)
-      .then((hash) => {
+    bcrypt
+      .hash(password, 10)
+      .then(hash => {
         this.password = hash;
         next();
       })
-      .catch((err) => next(err))
+      .catch(err => next(err));
   } else {
     next();
   }
 });
 
+UserSchema.statics.authenticate = function(email, password, callback) {
```

Comment:
This works. Bonus points for returning a promise with this! If you're up for a challenge, give it a try. I think it'd be healthy to let developers handle promises with methods like this instead of bouncing back and forth between callbacks and promises.

Reply:
Added mongoose `.exec()` method to return promise from User model.